core/core-configuration-layer.el

```diff
@@ -90,6 +96,32 @@ tarballs.")
   "Absolute path to the file containing the current stable elpa repository
 version")
 
+(defvar spacemacs--current-customization-group nil
+  "Used to auto-set `spacemacs|defc' group.")
+
+(defmacro spacemacs|defc (symbol standard doc type &optional group-override)
+  "Spacemacs flavored `defcustom' for configuring layers.
+SYMBOL         is the variable name; it should not be quoted.
+STANDARD       is an expression specifying the variable's standard value.
+DOC            is a doc-string.
+TYPE           should be a widget type for editing the symbol's value.
+               See Info node `(elisp) Customization Types' for a list of
+               base types and useful composite types.
+GROUP-OVERRIDE should be provided if you don't want Spacemacs to infer the
+               configuration group from the currently configured layer name."
+  (declare (indent defun) (doc-string 3) (debug (name body)))
+  `(progn
+     (put ',symbol 'spacemacs--defcustom-variable t)
+    (custom-declare-variable
+     ',symbol
+     ;; Took this from the `defcustom' implementation.
+     ,(if lexical-binding
+          ``(funcall #',(lambda () ,standard))
+        `',standard)
+     ,doc
+     :type ,type
+     :group ,(or group-override (quote spacemacs--current-customization-group)))))
```

Comment:
should default to 'spacemacs group.  Also should dry to infer the group based on buffer's file  (in case of buffer evaluation)